datasets/electricity/electricity.py

```diff
@@ -0,0 +1,217 @@
+# coding=utf-8
+# Copyright 2020 The HuggingFace Datasets Authors and the current dataset script contributor.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Electricity time series dataset."""
+
+
+from pathlib import Path
+
+import pandas as pd
+
+import datasets
+
+from .utils import to_dict
+
+
+_CITATION = """\
+@inproceedings{10.1145/3209978.3210006,
+    author = {Lai, Guokun and Chang, Wei-Cheng and Yang, Yiming and Liu, Hanxiao},
+    title = {Modeling Long- and Short-Term Temporal Patterns with Deep Neural Networks},
+    year = {2018},
+    isbn = {9781450356572},
+    publisher = {Association for Computing Machinery},
+    address = {New York, NY, USA},
+    url = {https://doi.org/10.1145/3209978.3210006},
+    doi = {10.1145/3209978.3210006},
+    booktitle = {The 41st International ACM SIGIR Conference on Research & Development in Information Retrieval},
+    pages = {95--104},
+    numpages = {10},
+    location = {Ann Arbor, MI, USA},
+    series = {SIGIR '18}
+}
+"""
+
+# You can copy an official description
+_DESCRIPTION = """\
+This new dataset contains hourly kW electricity consumption time series of 370 Portuguese clients from 2011 to 2014.
+"""
+
+_HOMEPAGE = "https://archive.ics.uci.edu/ml/datasets/ElectricityLoadDiagrams20112014"
+
+_LICENSE = ""
+
+# The HuggingFace Datasets library doesn't host the datasets but only points to the original files.
+# This can be an arbitrary nested dict/list of URLs (see below in `_split_generators` method)
+_URL = "https://archive.ics.uci.edu/ml/machine-learning-databases/00321/LD2011_2014.txt.zip"
+
+
+class Electricty(datasets.GeneratorBasedBuilder):
+    """Hourly electricity consumption of 370 points/clients."""
+
+    VERSION = datasets.Version("1.0.0")
+
+    BUILDER_CONFIGS = [
+        datasets.BuilderConfig(
+            name="uci",
+            version=VERSION,
+            description="Original UCI time series.",
+        ),
+        datasets.BuilderConfig(
+            name="lstnet",
+            version=VERSION,
+            description="Electricity time series preporcessed as in LSTNet paper.",
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = "lstnet"
+
+    def _info(self):
+        features = datasets.Features(
+            {
+                "start": datasets.Value("string"),
```

Comment:
Let's use the timestamp type here (you can find my message on Slack where I explain how to work with timestamps  in `datasets`):
```suggestion
                "start": datasets.Value("timestamp[<unit]"),
```
Please also update the README after that change.